### PR TITLE
AA-124: Show course end alert

### PR DESCRIPTION
**Pull request by mikix**
This is a parity-with-LMS change, to bring their warning about the course ending within a couple weeks to the MFE.

These new designs were approved by Hilary and Eugene.

### **Old view:**
Screenshot from 2020-07-30 14-28-14

### **New view (ending today):**
Screenshot from 2020-07-30 14-28-00

### **New view (ending days away):**
Screenshot from 2020-07-30 14-54-57

### CODE REVIEW

File: src/alerts/course-end-alert/CourseEndAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  const timezoneFormatArgs = userTimezone ? { timeZone: userTimezone } : {};
+
+  let msg;
+  if (delta < DAY_MS) {
+    // This weird "replace a format key with its own string again" code is so that react-intl messages look like normal
+    // to translators, but that react-intl doesn't yell at us for not providing format arguments.
+    msg = intl.formatMessage(messages.shortDelta, { courseEndTime: '{courseEndTime}', timeRemaining: '{timeRemaining}' });
+    msg = reactStringReplace(msg, '{courseEndTime}', () => (
+      <FormattedTime
```
Comment (mikix) [inline src/alerts/course-end-alert/CourseEndAlert.jsx]: @davidjoy is there a better way to do this? This feels hacky. Certainly hacky enough to require a comment. You mentioned that react-intl had some way of putting a component inside a string already?

File: package.json [hunk 2]
```diff
@@ -55,6 +55,7 @@
     "react-router": "^5.1.2",
     "react-router-dom": "^5.1.2",
     "react-share": "^4.1.0",
+    "react-string-replace": "^0.4.4",
```
Comment (mikix) [inline package.json]: Seems semi-widely used, with 82 dependents on npm.

**Comment by mikix** [comment]
@ekangedx and @hilaryg123: just FYI that this is landing. Screenshots in description above, but looks exactly like I already sent ya.

**Review by ciduarte — APPROVED** [review]
LGTM, thanks for explaining the order of ops here

**Pull request merged**